openedx/core/djangoapps/public_api/courses/tests/test_course_import_export.py

```diff
@@ -0,0 +1,416 @@
+        result = client.get(self.url, HTTP_ACCEPT='application/json')
+
+        self.assertListEqual(
+            json.loads(result.content)['courses'],
+            [unicode(self.course.location.course_key)]
+        )
+
+
+@override_settings(CONTENTSTORE=TEST_DATA_CONTENTSTORE)
+class ImportTestCase(CourseTestCase):
+    """
+    Unit tests for importing a course
+    """
+    def setUp(self):
+        super(ImportTestCase, self).setUp()
+        self.url = course_url('course_import_export_handler', self.course)
+        self.content_dir = path(tempfile.mkdtemp())
+
+        def touch(name):
+            """ Equivalent to shell's 'touch'"""
+            with file(name, 'a'):
+                os.utime(name, None)
+
+        # Create tar test files -----------------------------------------------
+        # OK course:
+        good_dir = tempfile.mkdtemp(dir=self.content_dir)
+        # test course being deeper down than top of tar file
+        embedded_dir = os.path.join(good_dir, "grandparent", "parent")
+        os.makedirs(os.path.join(embedded_dir, "course"))
+        with open(os.path.join(embedded_dir, "course.xml"), "w+") as f:
+            f.write('<course url_name="2013_Spring" org="EDx" course="0.00x"/>')
+
+        with open(
+                os.path.join(embedded_dir, "course", "2013_Spring.xml"),
+                "w+"
+        ) as f:
+            f.write('<course></course>')
+
+        self.good_tar = os.path.join(self.content_dir, "good.tar.gz")
+        with tarfile.open(self.good_tar, "w:gz") as gtar:
+            gtar.add(good_dir)
+
+        # Bad course (no 'course.xml' file):
+        bad_dir = tempfile.mkdtemp(dir=self.content_dir)
+        touch(os.path.join(bad_dir, "bad.xml"))
+        self.bad_tar = os.path.join(self.content_dir, "bad.tar.gz")
+        with tarfile.open(self.bad_tar, "w:gz") as btar:
+            btar.add(bad_dir)
+
+        self.unsafe_common_dir = path(tempfile.mkdtemp(dir=self.content_dir))
+
+    def tearDown(self):
+        shutil.rmtree(self.content_dir)
+
+    def test_no_coursexml(self):
+        """
+        Check that the response for a tar.gz import without a course.xml is
+        correct.
+        """
+        with open(self.bad_tar) as btar:
+            resp = self.client.post(
+                self.url,
+                {
+                    "name": self.bad_tar,
+                    "course-data": [btar]
+                })
+        self.assertEquals(resp.status_code, 415)
+        # Check that `ImportStatus` returns the appropriate stage (i.e., the
+        # stage at which import failed).
+        resp_status = self.client.get(
+            course_url(
+                'course_import_status_handler',
+                self.course,
+                kwargs={'filename': os.path.split(self.bad_tar)[1]}
+            )
+        )
+
+        self.assertEquals(json.loads(resp_status.content)["ImportStatus"], -2)
```

Comment:
Might want to break this up, so that if this test fails, it's easier to see where it fails.
```python
obj = json.loads(resp_status.content)
self.assertIn("ImportStatus", obj)
self.assertEquals(obj["ImportStatus"], -2)
```